example/lib/screens/preference_center.dart

```diff
@@ -0,0 +1,250 @@
+    return SwitchListTile(
+        title: Text('${item.display.title}', style:TextStyle(fontWeight: FontWeight.bold)),
+        subtitle: Text('${item.display.subtitle}'),
+        value: isSubscribedChannelSubscription(item.subscriptionId),
+        onChanged: (bool value) {
+          onPreferenceChannelItemToggled(item.subscriptionId, value);
+        }
+    );
+  }
+
+
+  Widget bindContactSubscriptionItem(PreferenceCenterContactSubscriptionItem item) {
+    return SwitchListTile(
+        title: Text('${item.display.title}', style:TextStyle(fontWeight: FontWeight.bold)),
+        subtitle: Text('${item.display.subtitle}'),
+        value: isSubscribedContactSubscription(item.subscriptionId, []),
+        onChanged: (bool value) {
+          onPreferenceContactSubscriptionItemToggled(item.subscriptionId, [], value);
+        },
+    );
+  }
+
+  List<String> scopesFromComponents (List<ChannelScope> scopes) {
+    List<String> finalList = [];
+    for (ChannelScope scope in scopes) {
+      finalList.add(scope.toString().split('.').last);
+    }
+    return finalList;
+  }
+
+  List<Widget> contactScopes(PreferenceCenterContactSubscriptionGroupItem item) {
+    List<PreferenceCenterContactSubscriptionGroupItemComponent> components = item.components;
+    List<Widget> widgets = [];
+    for (PreferenceCenterContactSubscriptionGroupItemComponent component in components) {
+      String componentLabel = component.display.title;
+      List<String> scopes = scopesFromComponents(component.scopes);
+      Widget widget = FilterChip(
+        avatar: CircleAvatar(
+          backgroundColor: Colors.grey.shade800,
+        ),
+        label: Text('$componentLabel'),
+        selected: isSubscribedContactSubscription(item.subscriptionId, scopes),
+        onSelected: (bool value) {
+          onPreferenceContactSubscriptionItemToggled(item.subscriptionId, scopes, value);
+        },
+      );
+      widgets.add(widget);
+    }
+    return widgets;
+  }
+
+  Widget bindContactSubscriptionGroupItem(PreferenceCenterContactSubscriptionGroupItem item) {
+    return Column(
+        children: <Widget>[
+          ListTile(
+            title: Text('${item.display.title}', style:TextStyle(fontWeight: FontWeight.bold)),
+            subtitle: Text('${item.display.subtitle}'),
+          ),
+          Wrap(
+            runAlignment : WrapAlignment.start,
+            spacing: 10,
+            children: contactScopes(item),
+          ),
+        ],
+    );
+  }
+
+  Widget bindAlertItem(PreferenceCenterAlertItem item) {
+    return ListTile(
+        title: Text('${item.display.title}', style:TextStyle(fontWeight: FontWeight.bold)),
+        subtitle: Text('${item.display.subtitle}'),
+    );
+  }
+
+  Widget item(IndexPath indexPath) {
+    List<PreferenceCenterItem> items = preferenceCenterConfig.sections[indexPath.section].items;
+    PreferenceCenterItem item = items[indexPath.item];
+    switch (item.type) {
```

Comment:
CI is complaining about a missing return for this method if we don't match any of the cases. Might need to throw or return a null that we can ignore higher up?